docs/getting_started.md

```diff
@@ -0,0 +1,169 @@
+
+
+## Setting up the environment
+
+### Installing fondant
+
+We suggest that you use a [virtual environment](https://docs.python.org/3/library/venv.html) for you project. Fondant supports Python >=3.8.
+
+Fondant can be installed using pip:
+
+```bash
+pip install fondant
+```
+
+For the latest development version, you might want to install from source instead:
+```bash
+pip install git+https://github.com/ml6team/fondant.git
+```
+You can validate the installation of fondant by running its root CLI command:
+```bash
+fondant --help
+```
+
+## Your first pipeline
+
+Create a `pipeline.py` file in the root of your project and add the following code:
+
+```Python
+from fondant.pipeline import Pipeline, ComponentOp
+
+my_pipeline = Pipeline(
+    pipeline_name='my_pipeline',
+    base_path='/home/username/my_pipeline',
+    pipeline_description='This is my pipeline',
+)
+```
+
+This is all you need to initialize a fondant pipeline:
+
+- **pipeline_name**: A name to reference your pipeline.
+- **base_path**: The base path of your pipeline, this will be used to store artifacts: the data between steps and the pipeline manifests. This base_path can be a local path or a cloud storage path (e.g. s3://my_bucket/artifacts, or gs://my_bucket/artifacts).
+- **pipeline_description**: A description of your pipeline.
+
+## Adding components
+
+Now that we have a pipeline, we can add components to it. Components are the building blocks of your pipeline. They are the individual steps that will be executed in your pipeline. There are 2 main types of components:
+
+- reusable components: These are components that are already created by the community and can be easily used in your pipeline. You can find a list of reusable components [here](https://github.com/ml6team/fondant/tree/main/components). They often have arguments that you can set to configure them for your use case.
+
+- custom components: These are the components you create to solve your use case. A custom component can be easily created by adding a `fondant_component.yaml`, `dockerfile` and `main.py` file to your component subdirectory. This file contains the metadata of your component. You can find more information about the `fondant_component.yaml` file [here](https://github.com/ml6team/fondant/blob/main/docs/component_spec.md)
+
+Let's add a reusable component to our pipeline. We will use the `load_from_hf_hub` component to read data from huggingface. Add the following code to your `pipeline.py` file:
+
+```Python
+load_from_hf_hub = ComponentOp.from_registry(
+    name='load_from_hf_hub',
+    component_spec_path='components/load_from_hf_hub/component_spec.yml',
+    arguments={
+        'dataset_name': 'huggan/pokemon',
+        'n_rows_to_load': 100,
+        'column_name_mapping': {
+            'image': 'images_raw',
+        },
+        "image_column_names": ["image"],
+        
+    }
+)
+
+my_pipeline.add_op(load_from_hf_hub, dependencies=[])
+```
```

Comment:
Not sure if it is the best example since it seems quite complex

Reply:
We might just have to explain it a bit better, as it is the most useful component to easily load some data to get started. Not sure if we want to describe this as a third kind of component, which is a "generic" component.